kubernetes/structure_endpointslice.go

```diff
@@ -0,0 +1,209 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package kubernetes
+
+import (
+	"strconv"
+
+	v1 "k8s.io/api/core/v1"
+	api "k8s.io/api/discovery/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func expandEndpointSliceEndpoints(in []interface{}) []api.Endpoint {
+	if in == nil || len(in) == 0 {
+		return []api.Endpoint{}
+	}
+	endpoints := make([]api.Endpoint, len(in))
+	for i, endpoint := range in {
+		r := api.Endpoint{}
+		endpointConfig := endpoint.(map[string]interface{})
+		if v := endpointConfig["addresses"].([]interface{}); len(v) != 0 {
+			r.Addresses = expandStringSlice(v)
+		}
+		if v, ok := endpointConfig["condition"].([]interface{}); ok {
+			r.Conditions = expandEndpointSliceCondition(v)
+		}
+		if v, ok := endpointConfig["hostname"].(string); ok && v != "" {
+			r.Hostname = ptrToString(v)
+		}
+		if v, ok := endpointConfig["node_name"].(string); ok && v != "" {
+			r.NodeName = ptrToString(v)
+		}
+		if v, ok := endpointConfig["target_ref"].([]interface{}); ok && len(v) != 0 {
+			r.TargetRef = expandObjectReference(v)
+		}
+		if v, ok := endpointConfig["zone"].(string); ok && v != "" {
+			r.Zone = ptrToString(v)
+		}
+
+		endpoints[i] = r
+	}
+	return endpoints
+}
+
+func expandObjectReference(l []interface{}) *v1.ObjectReference {
+	if len(l) == 0 || l == nil {
+		return &v1.ObjectReference{}
+	}
+	in := l[0].(map[string]interface{})
+	obj := &v1.ObjectReference{}
+
+	if v, ok := in["name"].(string); ok {
+		obj.Name = v
+	}
+	if v, ok := in["namespace"].(string); ok {
+		obj.Namespace = v
+	}
+	if v, ok := in["resource_version"].(string); ok {
+		obj.ResourceVersion = v
+	}
+	if v, ok := in["uid"]; ok {
+		obj.UID = types.UID(v.(string))
+	}
+	if v, ok := in["field_path"].(string); ok {
+		obj.FieldPath = v
+	}
+
+	return obj
+}
+
+func expandEndpointSlicePorts(in []interface{}) []api.EndpointPort {
+	if in == nil || len(in) == 0 {
+		return []api.EndpointPort{}
+	}
+	ports := make([]api.EndpointPort, len(in))
+	for i, port := range in {
+		r := api.EndpointPort{}
+		portCfg := port.(map[string]interface{})
+		if v, ok := portCfg["name"].(string); ok {
+			r.Name = ptrToString(v)
+		}
+		if v, ok := portCfg["port"].(string); ok {
+			if v == "" {
+				continue
+			}
+			v, _ := strconv.Atoi(v)
```

Comment:
Elsewhere we use ParseInt and set the size to 32 bits – this should remove the need for the bounds check below.

https://github.com/hashicorp/terraform-provider-kubernetes/blob/557097e7f6d6c46819edac76db46636095b35115/kubernetes/structure_job.go#L112-L116